src/main/java/net/sf/jabref/gui/mergeentries/MergeEntryDOIDialog.java

```diff
@@ -50,46 +51,75 @@
     private final BasePanel panel;
     private final CellConstraints cc = new CellConstraints();
     private BibEntry originalEntry;
-    private BibEntry doiEntry;
+    private BibEntry fetchedEntry;
     private NamedCompound ce;
     private MergeEntries mergeEntries;
 
     private final DOItoBibTeXFetcher doiFetcher = new DOItoBibTeXFetcher();
+    private final ISBNtoBibTeXFetcher isbnFetcher = new ISBNtoBibTeXFetcher();
 
     private static final String MARGIN = "5px";
 
+    private static String fetchedString;
 
     public MergeEntryDOIDialog(BasePanel panel) {
-        super(panel.frame(), Localization.lang("Merge entry with DOI information"), true);
+        super(panel.frame(), Localization.lang("Merge entry with DOI/ISBN information"), true);
 
         this.panel = panel;
 
         if (panel.getSelectedEntries().size() != 1) {
             JOptionPane.showMessageDialog(panel.frame(),
                     Localization.lang("This operation requires exactly one item to be selected."),
-                    Localization.lang("Merge entry with DOI information"), JOptionPane.INFORMATION_MESSAGE);
+                    Localization.lang("Merge entry with DOI/ISBN information"), JOptionPane.INFORMATION_MESSAGE);
             this.dispose();
             return;
         }
 
         this.originalEntry = panel.getSelectedEntries().get(0);
-        panel.output(Localization.lang("Fetching info based on DOI"));
         Optional<String> doi = this.originalEntry.getFieldOptional(FieldName.DOI);
```

Comment:
I would restructure the flow a bit: The dialog just gets two BibEntries which should be compared/merged. So it's the callers responsibility to provide the second entry (i.e. fetch from ISBN / DOI /...)